presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ActualProperties.java

```diff
@@ -367,38 +411,39 @@ private boolean isNodePartitionedWith(
             return nodePartitioning.map(Partitioning::getPartitioningArguments);
         }
 
-        private boolean isStreamPartitionedOn(Collection<Symbol> columns, Set<Symbol> constants)
+        private boolean isStreamPartitionedOn(Collection<Symbol> columns, Set<Symbol> constants, boolean nullsReplicated)
         {
-            return streamPartitioning.isPresent() && streamPartitioning.get().isPartitionedOn(columns, constants);
+            return streamPartitioning.isPresent() && streamPartitioning.get().isPartitionedOn(columns, constants) && this.nullsReplicated == nullsReplicated;
         }
 
         /**
          * @return true if all the data will effectively land in a single stream
          */
         private boolean isEffectivelySingleStream(Set<Symbol> constants)
         {
-            return streamPartitioning.isPresent() && streamPartitioning.get().isEffectivelySinglePartition(constants);
+            return streamPartitioning.isPresent() && streamPartitioning.get().isEffectivelySinglePartition(constants) && !nullsReplicated;
         }
 
         /**
          * @return true if repartitioning on the keys will yield some difference
          */
         private boolean isStreamRepartitionEffective(Collection<Symbol> keys, Set<Symbol> constants)
         {
-            return !streamPartitioning.isPresent() || streamPartitioning.get().isRepartitionEffective(keys, constants);
+            return (!streamPartitioning.isPresent() || streamPartitioning.get().isRepartitionEffective(keys, constants)) && !nullsReplicated;
```

Comment:
I'm not sure about this one... if you did have nulls replicated and you repartitioned, it would yield a difference, but that is likely not what someone wants.